source/common/network/tcp_listener_impl.cc

```diff
@@ -40,8 +40,9 @@ void TcpListenerImpl::onSocketEvent(short flags) {
   ASSERT(bind_to_port_);
   ASSERT(flags & (Event::FileReadyType::Read));
 
-  // TODO(fcoras): Add limit on number of accepted calls per wakeup
-  while (1) {
+  uint32_t connections_accepted_count = 0;
+  for (; connections_accepted_count < max_connections_to_accept_per_socket_event_;
```

Comment:
Seems that the loop may reject a connection. In that case, there may be 0 accepted connections, but the loop will end due to `connections_accepted_count == max_connections_to_accept_per_socket_event_`.
Should probably only increase the `connections_accepted_count` only when a connection is accepted.

Reply:
Thanks for raising this (and the other comment below). I will rename `connections_accepted_count` to `connections_accepted_from_kernel_count` to better capture what this is doing so it's more clear.